vsphere_plugin_common/__init__.py

```diff
@@ -1815,11 +1875,45 @@ def create_server(
                 "Error during executing VM creation task. VM name: \'{0}\'."
                 .format(vm_name))
 
-        vm = self._get_obj_by_name(
+        # VM object created. Now perform final post-creation tasks
+        vm = task.info.result
+        logger().info('VM version: vmx-{old}/vmx-{new}'.format(
+            old=self._get_virtual_hardware_version(vm),
+            new=minimal_vm_version))
+        if self._get_virtual_hardware_version(vm) < minimal_vm_version:
+            if enable_start_vm:
+                logger().info("Use VM hardware update with `enable_start_vm` "
+                              "is unsupported.")
+            else:
+                logger().info("Going to update VM hardware version.")
+                try:
+                    self._wait_for_task(vm.UpgradeVM_Task(
+                        "vmx-{version}".format(version=minimal_vm_version)))
+                except Exception as e:
```

Comment:
Do we know what possible exceptions there are?

Reply:
Replaced to correct exception.